### DISCUSSION TITLE
Call for tunning results on CLBlast to achieve faster llama.cpp prompt performance

### DISCUSSION THREAD

**Discussion by tangjinchuan** [ggml-org/llama.cpp#1688] [General]
Dear all,

Please forgive me if this seems to be spam.

Since llama.cpp can run on non-cuda GPUs with the help of CLBlast, each GPU could have a different architecture that needs different parameters to achieve the best matrix multiplication. In many cases, CLBlast is a library that can achieve higher performance with GEMM after tunning. I am a volunteer to help this CLBlast project to find different tuning results for different GPUs, it would be very great if you could kindly run CLBlast tuner on your GPU (all GPUs are very welcome, the tutorial is given as follows ) and report the tunning results in zip file to CLBlast site here:

[New tuning results · Issue #1 · CNugteren/CLBlast · GitHub](https://github.com/CNugteren/CLBlast/issues/1)

This could help to make CLBlast and llama.cpp faster!!!

Official manual on running the tuner (especially for Linux/MacOS users):

[CLBlast/tuning.md at master · CNugteren/CLBlast · GitHub](https://github.com/CNugteren/CLBlast/blob/master/doc/tuning.md)

My thread and file for running the tuner easily in Windows:

https://github.com/CNugteren/CLBlast/issues/1#issuecomment-1570253475

Best wishes,

Jinchuan Tang

**Comment by ghost** [comment]
Thanks for the information, I ran tuning and added the results for Adreno 640, which is now merged with CLBlast.
